controller/api/destination/watcher/endpoints_watcher.go

```diff
@@ -1050,6 +1083,33 @@ func (pp *portPublisher) newPodRefAddress(endpointPort Port, endpointIP, podName
 	return addr, id, nil
 }
 
+func (pp *portPublisher) newExtRefAddress(endpointPort Port, endpointIP, externalWorkloadName, externalWorkloadNamespace string) (Address, ExternalWorkloadID, error) {
+	id := ExternalWorkloadID{
+		Name:      externalWorkloadName,
+		Namespace: externalWorkloadNamespace,
+	}
+
+	ew, err := pp.k8sAPI.ExtWorkload().Lister().ExternalWorkloads(id.Namespace).Get(id.Name)
+	if err != nil {
+		return Address{}, ExternalWorkloadID{}, fmt.Errorf("unable to fetch ExternalWorkload %v: %w", id, err)
+	}
+
+	ownerKind, ownerName, err := pp.metadataAPI.GetOwnerKindAndName(context.Background(), ew.Kind, ew, false)
```

Comment:
Does this make sense for external workloads?  External workloads don't necessarily have an owner ref, and they certainly won't have the same owner ref kinds as pods (deployment, replicaset, job, etc.).  I'd be inclined to leave out the changes to GetOwnerKindAndName since we know that the external workload owner ref isn't going to match any of our known kinds.